src/main/java/org/terasology/gookeeper/system/EconomySystem.java

```diff
@@ -111,10 +217,14 @@ public void update(float delta) {
      */
     @Override
     public void payEntranceFee (EntityRef visitor) {
-        for (EntityRef wallet : entityManager.getEntitiesWith(EconomyComponent.class)) {
-            EconomyComponent economyComponent = wallet.getComponent(EconomyComponent.class);
+        VisitorComponent visitorComponent = visitor.getComponent(VisitorComponent.class);
+        VisitorEntranceComponent visitorEntranceComponent = visitorComponent.visitorEntranceBlock.getComponent(VisitorEntranceComponent.class);
+        EntityRef player = visitorEntranceComponent.owner;
```

Comment:
Have you explored the built in owner functionality for entities? That seems like it might also work